test/e2e/service_test.go

```diff
@@ -106,34 +106,47 @@ func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name st
 	return svc, cleanup
 }
 
-// TestNodePortWindows tests NodePort Service on Windows Node. It is a temporary test to replace upstream Kubernetes one:
-// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
-// Issue: https://github.com/antrea-io/antrea/issues/2289
-func TestNodePortWindows(t *testing.T) {
-	skipIfNoWindowsNodes(t)
-
+// TestNodePort tests traffic from Pods to NodePort Service.
+func TestNodePort(t *testing.T) {
 	data, err := setupTest(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 
+	nodeIP := clusterInfo.nodes[0].ip()
+	t.Run("NodePort", func(t *testing.T) {
+		t.Run("Windows NodePort Service", func(t *testing.T) {
+			// Windows NodePort Service: It is a temporary test to replace upstream Kubernetes one:
+			// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
+			// Issue: https://github.com/antrea-io/antrea/issues/2289
+			// Unlike upstream Kubernetes Conformance, the client is on a Linux Node (nodeName(0)).
+			// It doesn't need to be the control-plane for e2e test and other Linux workers will work as well. However, in this
+			// e2e framework, nodeName(0)/Control-plane Node is guaranteed to be a Linux one.
+			skipIfNoWindowsNodes(t)
+			svcNode := nodeName(clusterInfo.windowsNodes[0])
+			testNodePort(t, data, svcNode, "windows-service-test", nodeIP)
+		})
+		t.Run("Linux NodePort Service", func(t *testing.T) {
+			svcNode := clusterInfo.nodes[0].name
+			testNodePort(t, data, svcNode, "linux-service-test", nodeIP)
+		})
+	})
```

Comment:
not sure I understand the nesting here. The first level (`t.Run("NodePort", func(t *testing.T) {`) seems unnecessary to me.

Reply:
Yes. Updated.